### PR TITLE
[stdlib] swapAt method

### PR DESCRIPTION
**Pull request by airspeedswift**

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please benchmark

**Comment by airspeedswift** [comment]
@swift-ci please test

File: stdlib/public/core/MutableCollection.swift [hunk 1]
```diff
@@ -372,6 +377,17 @@ extension MutableCollection {
       _writeBackMutableSlice(&self, bounds: bounds, slice: newValue)
     }
   }
+
+  /// Exchange the values at indices `i` and `j`.
+  ///
+  /// Has no effect when `i` and `j` are equal.
+  @_inlineable
+  public mutating func swapAt(_ i: Index, _ j: Index) {
+    guard i != j else { return }
```
Comment (xwu) [inline stdlib/public/core/MutableCollection.swift]: This is an interesting nuance. For reference types, this wouldn't be a drop-in replacement for the old `swap`, would it?

Comment (airspeedswift) [inline stdlib/public/core/MutableCollection.swift]: Well, the old `swap` used to (attempt to) trap in this case, so it should be. Unless I'm missing something.

**Comment by airspeedswift** [comment]
@swift-ci please test

File: stdlib/public/core/MutableCollection.swift (hunk 1 above)
Comment (xwu) [inline stdlib/public/core/MutableCollection.swift]: @airspeedswift My bad. You're checking for equality of the _indices_.

**Comment by airspeedswift** [comment]
@swift-ci please clean test

**Comment by airspeedswift** [comment]
@swift-ci please benchmark

**Comment by airspeedswift** [comment]
@swift-ci please clean test macOS platform

**Comment by airspeedswift** [comment]
Looks like OS X tests passed based on the logs, but the bot hung updating status.

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

**Pull request merged**